appservice/src/main/java/nu/yona/server/subscriptions/rest/PinResetRequestController.java

```diff
@@ -132,13 +132,13 @@
 
 	public void addLinks(UserResource userResource)
 	{
-		ConfirmationCode confirmationCode = userService.getUserEntityById(userResource.getContent().getId())
+		Optional<ConfirmationCode> confirmationCode = userService.getUserEntityById(userResource.getContent().getId())
 				.getPinResetConfirmationCode();
-		if (confirmationCode == null || pinResetRequestService.isExpired(confirmationCode))
+		if (!pinResetRequestService.isValidConfirmationCode(confirmationCode))
 		{
 			addPinResetRequestLink(userResource);
 		}
-		else if (confirmationCode.getCode() != null)
+		else if (confirmationCode.get().getCode() != null)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "confirmationCode.isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://10.127.10.107:9000/coding_rules#rule_key=squid%3AS3655)

Reply:
False positive. The check on line 137 verifies that the confirmation code is present.